src/WordsPosition.java

```diff
@@ -9,12 +9,18 @@ public WordsPosition(int x, int y) {
 	}
 	
 	public WordsPosition(double x, double y) {
-		this.x = (int) x;
-		this.y = (int) y;
+		this.x = (int) Math.round(x);
+		this.y = (int) Math.round(y);
 	}
 	
 	@Override 
 	public String toString() {
 		return Integer.toString(x) + " , " + Integer.toString(y);
 	}
+	
+	@Override
+	public boolean equals(Object other) {
+		WordsPosition otherPos = (WordsPosition) other;
```

Comment:
Technically, before casting, I think you should check if 'other' is an instance of WordsPosition.  Otherwise, the cast would result in an error.  We could even use EqualsBuilder, though I don't care too much about that part.  See this post: http://stackoverflow.com/questions/27581/what-issues-should-be-considered-when-overriding-equals-and-hashcode-in-java